### ISSUE TITLE
Different outputs for Windows and Mac / Linux

### ISSUE THREAD

**Issue by mothur-westcott** [torognes/vsearch#371]
Thanks for all your great work with vsearch! I work on the mothur project which includes a chimera.vsearch command. The command is a wrapper for your vsearch program. I've hit an issue with the parsing of vsearch's output files.  I am getting slightly different results for the MAC version and the Windows versions of vsearch and hope you can help.  I have attached the input file used with vsearch and the differing output files for your reference.  The command I am running is:

vsearch  --uchime_denovo algas2.trim.contigs.good.unique.good.filter.unique.precluster.tempFB01 --chimeras algas2.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.chimerasFB01vsearch_out --uchimeout algas2.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.chimerasFB01 --xsize --threads 4 

I am using vsearch version 2.11.1.

The MAC results look like:

>M04376_266_000000000-C84N4_1_1112_22889_19583
CTACGGGTGGCAGCAGTGAGGAATTTTCCGCAATGGGCGAAAGCC...

The Windows results look like:

>M04376_266_000000000-C84N4_1_1112_22889_19583;
CTACGGGTGGCAGCAGTGAGGAATTTTCCGCAATGGGCGAAAGCC...

The extra ";" attached to the sequence name.

Thanks for your help,
  Sarah

[vsearchFIles.zip](https://github.com/torognes/vsearch/files/3154546/vsearchFIles.zip)

**Comment by torognes** [comment]
Thanks for reporting this problem. I have been able to reproduce it.

This error is caused by incorrect parsing of FASTA/Q headers on Windows due to the differences in newline characters on Windows vs Mac/Linux (CR LF vs LF). The problem was solved in version 2.13.3. I have confirmed that the output is correct with that version.

All commands that write FASTA/Q files may be affected.

**Comment by frederic-mahe** [comment]
I've added tests to cover that issue (https://github.com/frederic-mahe/vsearch-tests/commit/3b5cbf101eb565d680c99ad673fae8e7ec2f4428). I can confirm that the bug can be replicated with older vsearch versions, but vsearch 2.13.3 passes the test:
```sh
printf "@s;size=1;\r\nA\n+\nI\n" | vsearch --fastx_filter - --xsize --quiet --fastqout -
```
I'll go ahead and close that issue. Thanks for the report @mothur-westcott
